src/components/ResourceList/components/FilterControl/components/DateSelector/DateSelector.tsx

```diff
@@ -408,8 +413,22 @@ function getDateFilterOption(
   return filterValue;
 }
 
-function formatDateValue(date: Date) {
-  return date.toISOString().slice(0, 10);
+function formatDateForTimezone(date: Date) {
+  const hourOffset = date.getTimezoneOffset() / 60;
+
+  if (hourOffset === 0) {
+    return date.toISOString();
+  }
+
+  const midnight =
+    hourOffset > -1
+      ? date.getHours() + hourOffset
+      : date.getHours() - hourOffset;
+  const updatedDate = new Date(date.setHours(midnight));
+  const month = padStart(String(updatedDate.getUTCMonth() + 1), 2, '0');
```

Comment:
```suggestion
  const month = `0${String(updatedDate.getUTCMonth() + 1)}`.slice(-2);
```

I think `padStart` is going to do [more work than we need it to](https://github.com/lodash/lodash/blob/4.17.10/lodash.js#L14442).